### ISSUE TITLE
"coinselect" always expose the largest UTXO no matter how small the payment is because it sorts by decending value.

### ISSUE THREAD

**Issue by Transisto** [bitcoinjs/coinselect#74]
This is a huge privacy issue.

It's exposing a larger balance than required to the recipient.

**Comment by LeeKorbisCa** [comment]
@Transisto I noticed this also, but I think it has to do with minimizing the final fee, since more UTXO = higher fee?

I'm about to use this in production so I'd greatly appreciate if you have more opinions around this.

For context: as a service, privacy isnt a goal for ourselves, but reduced fee costs.

**Comment by nostitos** [comment]
There's more details in the Bluewallet issue above. 

I've stopped using bluewallet mostly for that reason.

**Comment by Transisto** [comment]
There's more details in the Bluewallet issue above.

I've stopped using bluewallet mostly for that reason.

It's not using less fee if it's using a 10btc vs a 1btc utxo to send 0.1

**Comment by LeeKorbisCa** [comment]
@Transisto what I meant was, if it were the other way around, maybe it'd use many small UTXOs - but yeah, obviously there are ways to use "the least largest, satisfactory utxo". Maybe it's more effective to use the largest UTXO as many times as possible before using the small ones? It's hard to say. @Overtorment what do you think or know about this?

**Comment by Transisto** [comment]
Sure, if you're running an exchange you might not care at all about
privacy, go ahead and use this.

Pay 100$ with a 10,000$ utxo in a third world country and you have a high
chance of getting robbed.
Then spend 3 months spending from the same UTXO, creating a long and
obvious peel chain,

Privacy is already bad enough in Bitcoin, this coin selection algo just
make it much worst.

**Comment by LeeKorbisCa** [comment]
Yes I completely agree. For an individual the current algorithms are not
suitable. It seems though they are very open to accepting PRs that could
remedy this problem :D.

Not only code is needed, but more documentation around usage. Otherwise
people won't know what to use.

Thank you for further explaining things :)
